javascript/ql/src/Security/CWE-327/BadRandomness.ql

```diff
@@ -99,10 +150,13 @@ DataFlow::Node badCrypto(string description, DataFlow::SourceNode source) {
     )
   )
   or
-  // division - always bad
+  // division - bad if result is rounded.
   exists(DivExpr div | result.asExpr() = div |
     goodRandom(source).asExpr() = div.getLeftOperand() and
-    description = "division"
+    description = "division and rounding the result" and
+    not div.getRightOperand() = isPowerOfTwoMinusOne().asExpr() and // division by (2^n)-1 most of the time produces a uniformly random number between 0 and 1.
+    div.getParentExpr+() =
```

Comment:
This is a minimal (and heuristic!) taint analysis that covers the cases like `Math.round(... ( ... ( x / y ) ... ) ... )`, right?

Reply:
Yes. 
For the CVE the `+` is not needed. But I can easily imagine some case the division is nested inside some other arithmetic. 

However, the `+` is not needed for any of the results on LGTM, so I can remove it if you prefer.

Reply:
I worry that we flag the inter-procedural case: `Math.round(doStuff(x/y))` as a weird FP.

If we do not need the `+`, then I would prefer to use `round.getArgument(0)` instead of `getParent()`.